test/hotspot/jtreg/runtime/Monitor/libIterateMonitorWithDeadObjectTest.c

```diff
@@ -0,0 +1,142 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <jni.h>
+#include <stdlib.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#define die(x) do { printf("%s:%s\n",x , __func__); perror(x); exit(EXIT_FAILURE); } while (0)
+
+#ifndef _Included_IterateMonitorWithDeadObjectTest
+#define _Included_IterateMonitorWithDeadObjectTest
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+static JavaVM* jvm;
+static pthread_t attacher;
+
+static jobject create_object(JNIEnv* env) {
+  jclass clazz = (*env)->FindClass(env, "java/lang/Object");
+  if (clazz == 0) die("No class");
+
+  jmethodID constructor = (*env)->GetMethodID(env, clazz, "<init>", "()V");
+  if (constructor == 0) die("No constructor");
+
+  return (*env)->NewObject(env, clazz, constructor);
+}
+
+static void system_gc(JNIEnv* env) {
+  jclass clazz = (*env)->FindClass(env, "java/lang/System");
+  if (clazz == 0) die("No class");
+
+  jmethodID method = (*env)->GetStaticMethodID(env, clazz, "gc", "()V");
+  if (method == 0) die("No method");
+
+  (*env)->CallStaticVoidMethod(env, clazz, method);
+}
+
+static void thread_dump_with_locked_monitors(JNIEnv* env) {
+  jclass ManagementFactoryClass = (*env)->FindClass(env, "java/lang/management/ManagementFactory");
+  if (ManagementFactoryClass == 0) die("No ManagementFactory class");
+
+  jmethodID getThreadMXBeanMethod = (*env)->GetStaticMethodID(env, ManagementFactoryClass, "getThreadMXBean", "()Ljava/lang/management/ThreadMXBean;");
+  if (getThreadMXBeanMethod == 0) die("No getThreadMXBean method");
+
+  jobject threadBean = (*env)->CallStaticObjectMethod(env, ManagementFactoryClass, getThreadMXBeanMethod);
+
+  jclass ThreadMXBeanClass = (*env)->FindClass(env, "java/lang/management/ThreadMXBean");
+  if (ThreadMXBeanClass == 0) die("No ThreadMXBean class");
+
+  jmethodID dumpAllThreadsMethod = (*env)->GetMethodID(env, ThreadMXBeanClass, "dumpAllThreads", "(ZZ)[Ljava/lang/management/ThreadInfo;");
+  if (dumpAllThreadsMethod == 0) die("No dumpAllThreads method");
+
+  // The 'lockedMonitors == true' is what triggers the collection of the monitor with the dead object.
```

Comment:
"triggers the collection" sounds like a GC interaction but that is not what you mean. Suggestion:
```
// The 'lockedMonitors == true' is what causes the monitor with a dead object to be examined.
```